tree/treeplayer/inc/TMPWorkerTree.h

```diff
@@ -199,33 +199,37 @@ void TMPWorkerTreeFunc<F>::Process(UInt_t code, MPCodeBufPair &msg)
       return;
    }
 
+   // If we are not done processing entries in the tree, 
    // create a TTreeReader that reads this range of entries
-   TTreeReader reader(fTree, enl);
+   if (start != fTree->GetEntries()) {
```

Comment:
Maybe this instead?
```suggestion
   if (start >= 0 && start < fTree->GetEntries()) {
```
Unless `start < 0` has some meaning